rust/arrow/src/array/equal/boolean.rs

```diff
@@ -77,20 +77,14 @@ pub(super) fn boolean_equal(
         let rhs_null_bytes = rhs_nulls.as_ref().unwrap().as_slice();
 
         (0..len).all(|i| {
-            let lhs_pos = lhs_start + i;
-            let rhs_pos = rhs_start + i;
+            let lhs_pos = lhs.offset() + lhs_start + i;
+            let rhs_pos = rhs.offset() + rhs_start + i;
```

Comment:
There was a comment from @mqy about moving this out of the iter, as it was hurting performance.
It's unexpected that this wouldn't be hoisted out by the compiler, but IIWII.